From b44a616253e41d250c5a228444c8530c151d08cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timur=20Krist=C3=B3f?= <timur.kristof@gmail.com>
Date: Tue, 26 Sep 2023 11:53:00 +0200
Subject: [PATCH 548/834] ac: Clarify SDMA opcode defines.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Make the naming of these defines consistent with others.
And add a few new opcodes and related definitions.

Signed-off-by: Timur Kristóf <timur.kristof@gmail.com>
Reviewed-by: Samuel Pitoiset <samuel.pitoiset@gmail.com>
Reviewed-by: Marek Olšák <marek.olsak@amd.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25769>
---
 src/amd/common/sid.h | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/amd/common/sid.h b/src/amd/common/sid.h
index e82a2275e3a..03d4793fabb 100644
--- a/src/amd/common/sid.h
+++ b/src/amd/common/sid.h
@@ -337,15 +337,19 @@
 #define SDMA_WRITE_SUB_OPCODE_LINEAR               0x0
 #define SDMA_WRITE_SUB_OPCODE_TILED                0x1
 #define CIK_SDMA_OPCODE_INDIRECT_BUFFER            0x4
-#define CIK_SDMA_PACKET_FENCE                      0x5
-#define CIK_SDMA_PACKET_TRAP                       0x6
-#define CIK_SDMA_PACKET_SEMAPHORE                  0x7
-#define CIK_SDMA_PACKET_CONSTANT_FILL              0xb
+#define CIK_SDMA_OPCODE_FENCE                      0x5
+#define SDMA_FENCE_MTYPE_UC                        0x3
+#define CIK_SDMA_OPCODE_TRAP                       0x6
+#define CIK_SDMA_OPCODE_SEMAPHORE                  0x7
+#define CIK_SDMA_OPCODE_POLL_REGMEM                0x8
+#define SDMA_POLL_INTERVAL_160_CLK                 0xa
+#define SDMA_POLL_RETRY_INDEFINITELY               0xfff
+#define CIK_SDMA_OPCODE_CONSTANT_FILL              0xb
 #define CIK_SDMA_OPCODE_TIMESTAMP                  0xd
 #define SDMA_TS_SUB_OPCODE_SET_LOCAL_TIMESTAMP     0x0
 #define SDMA_TS_SUB_OPCODE_GET_LOCAL_TIMESTAMP     0x1
 #define SDMA_TS_SUB_OPCODE_GET_GLOBAL_TIMESTAMP    0x2
-#define CIK_SDMA_PACKET_SRBM_WRITE                 0xe
+#define CIK_SDMA_OPCODE_SRBM_WRITE                 0xe
 /* There is apparently an undocumented HW limitation that
    prevents the HW from copying the last 255 bytes of (1 << 22) - 1 */
 #define CIK_SDMA_COPY_MAX_SIZE    0x3fff00   /* almost 4 MB*/
-- 
2.42.0

